### ISSUE TITLE
Not being able to push onto git

### ISSUE THREAD

**Issue by jennybc** [STAT545-UBC/Discussion#99]
Moving out of email, opening on behalf of @vedangvyas

> As per the guidelines on the STAT545, I was able to pull the readme file from my private repo (albeit, I had to do it through shell for some odd reason). I made an edit and now I've been trying to push it back onto Git using the instructions provided. I saved and committed the changes, and (pulled for the habit of it) pushed onto Git. It doesn't give me an error but it also doesn't give any sort of confirmation. When I refresh my private repo page on Git, the changes I made do not appear. Is there a manual way I can do this (via shell perhaps)?

Please go through the various steps in [our guide to Git + GitHub + RStudio](http://stat545-ubc.github.io/git00_index.html), especially the one about [Test connection to GitHub](http://stat545-ubc.github.io/git05_github-connection.html). That will involve pushing from command line Git and should expose various problems. Also consider if you need to [cache credentials](http://stat545-ubc.github.io/git06_credential-caching.html). We can't tell if RStudio is the problem until you've gone through that.

> If you are not able to answer my query via email, when would be the most appropriate time for me to stop by either your office or the TAs' office for some assistance? 

Office hour issue: #85

**Comment by vedangvyas** [comment]
Turns out I had git 1.7.4.4 installed and it wouldn't update. I ended up uninstalling and reinstalling git and now the version is higher than 1.7.10. The problem now is that RStudio is unwilling to interact with git. The path to git on my computer is /opt/git/bin/git and I can't seem to change that Git executable in RStudio to that path. Is this suppose to happen?

**Comment by jennybc** [comment]
What exactly happens in RStudio when you do *Tools > Global Options ... > Git/SVN* and you attempt to put `/opt/git/bin/git` in the Git executable field and click "Apply"?

Does the tutorial stuff where you're using Git from the command line all work?

**Comment by vedangvyas** [comment]
I wasn't able to edit the Git executable at all :S. Not sure why that was the case. And yes, pushing using command line was working fine. 

For some odd reason, now terminal doesn't respond to the command which git. When I enter git --version, it says that git: command not found. I am certain I installed git.

**Comment by vedangvyas** [comment]
I just reinstalled git. which git gives me /usr/local/git/bin/git but when I check the git version (git --version), terminal says Illegal Instruction: 4.

**Comment by samhinshaw** [comment]
It may be related to the version of OS X you're running.

http://gitforums.org/forum/main-category/main-forum/1264-illegal-instruction-4-when-running-git-on-osx-10-7-5

http://stackoverflow.com/questions/14268887/what-is-the-illegal-instruction-4-error-and-why-does-mmacosx-version-min-10

**Comment by jennybc** [comment]
Nice demo @samhinshaw of the goodness of Googling your error message.

**Comment by vedangvyas** [comment]
Actually, the first link is the exact link I followed to get Git for my Mac OS Lion yesterday. I followed the instructions to uninstall then reinstall Git using the link on wandisco. That's when I thought (and checked) that a newer version of git had been installed. Thereafter, I was able to push using the command line and everything seemed to work properly. But when I closed terminal and reopened it this morning, it wouldn't recognize the command git --version (as in, it couldn't locate git I assume).

**Comment by vedangvyas** [comment]
I am going to start from the top again and do it once more. Perhaps I missed some intermediate steps which is causing these problems. Thanks for your help!

**Comment by jennybc** [comment]
You might want to ... log in and log out? Or restart. Just because you never know.

More concrete suggestion:

In the shell, what do you see when you enter this:

``` shell
echo $PATH
```

Specifically, is the location of the git executable in there somewhere?

**Comment by jennybc** [comment]
@vedangvyas I'd still really like to know about your `PATH`. I want to understand how/if `PATH` problems relate to RStudio not finding git on its own.

I assume once I left you were able to push? From RStudio?

**Comment by vedangvyas** [comment]
So I think the problem was that after installing Git, the problem was that shell couldn't find the path to where Git was. Each time I opened shell, I would have to enter export PATH=/opt/git/bin:$PATH. So every time I closed shell, it would forget the path to Git (I assume this was what was happening but I'm not really sure). Dean used his magic abilities and (what I assume he did) got shell to permanently note the path to Git. After that, using the Command+Shift+G in Git executable, I was able to enter the path to Git. And moreover, even when I closed shell, it would remember where Git was. 

I was also able to successfully cache my username and password so that I wouldn't have to enter them each time I wanted to push. And I was able to push from RStudio itself.
